src/Markup/Avalonia.Markup.Xaml/Data/Binding.cs

```diff
@@ -271,12 +222,13 @@ private ExpressionObserver CreateElementObserver(IControl target, string element
 
         private ExpressionObserver CreateFindAncestorObserver(
             IControl target,
+            RelativeSource relativeSource,
             string path)
         {
             Contract.Requires<ArgumentNullException>(target != null);
 
             return new ExpressionObserver(
-                ControlLocator.Track(target, RelativeSource.AncestorType, RelativeSource.AncestorLevel -1),
+                ControlLocator.Track(target, relativeSource.AncestorType != null ? relativeSource.Tree : TreeType.Logical, relativeSource.AncestorLevel - 1, relativeSource.AncestorType),
```

Comment:
One final thing: is `relativeSource.AncestorType != null ? relativeSource.Tree : TreeType.Logical` still needed here now that the visual tree can't be searched with no `AncestorType`?

Reply:
Good catch! I'll get rid of that.